### PR TITLE
:sparkles: Add locations service

### PR DESCRIPTION
**Pull request by gerbernoah**

### CODE REVIEW

File: src/services/locations_v1.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+const storeData = async (request: LocationRequest, env: Env): Promise<Response> => {
+  const id = request.from;
+  const location = request.timedLocation;
+
+  if (!location.latitude || !location.longitude || !location.altitude || !location.timestamp) {
+    throw new Response('Data wrong format', { status: 400 });
+  }
+
+  if (Date.now() < location.timestamp) {
```
Comment (7rebux) [inline src/services/locations_v1.ts:67]: This statement is meant to be flipped no?

File: src/services/locations_v1.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    throw new Response('Invalid data', { status: 400 });
+  }
+
+  //cut incomming data
+  const kvData: TimedLocation = {
```
Comment (7rebux) [inline src/services/locations_v1.ts:72]: Why is this needed

File: src/services/locations_v1.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if (!id) {
+          throw new Error('No user id provided');
+        }
+
+        const friends: string[] = (await env.FRIENDS_KV.get(senderId, 'json')) ?? [];
+
+        if (!friends.includes(id)) {
+          console.log(id);
+          console.log(friends);
+          return new Response('No access');
```
Comment (7rebux) [inline src/services/locations_v1.ts:49]: Whats the statusCode of this response?

File: src/services/locations_v1.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+        if (!id) {
+          throw new Error('No user id provided');
+        }
+
+        const friends: string[] = (await env.FRIENDS_KV.get(senderId, 'json')) ?? [];
+
+        if (!friends.includes(id)) {
+          console.log(id);
+          console.log(friends);
```
Comment (7rebux) [inline src/services/locations_v1.ts:48]: Are these logs needed?

**Review by 7rebux — CHANGES_REQUESTED** [review]

File: src/services/locations_v1.ts (hunk 1 above)
Comment (gerbernoah) [reply src/services/locations_v1.ts:67]: I dont think so. If the timestamp of the location is in the future (which is not possible) we return "Invalid data"

File: src/services/locations_v1.ts (hunk 2 above)
Comment (gerbernoah) [reply src/services/locations_v1.ts:72]: To ensure that only those 4 fields are stored in the Kv and no "hidden" fields are just passed through and stored aswell since the type cast doesnt cut the data stored

File: src/services/locations_v1.ts (hunk 1 above)
Comment (7rebux) [reply src/services/locations_v1.ts:65]: I think `location.timestamp > Date.now()` is way better readable

File: src/services/locations_v1.ts (hunk 2 above)
Comment (7rebux) [reply src/services/locations_v1.ts:70]: fair enough

**Review by 7rebux — APPROVED** [review]
